deploy/templates/service-template.yml

```diff
@@ -346,56 +281,48 @@ objects:
                         domains:
                           - "*"
                         routes:
-                        - match: 
-                            prefix: /
+                        - match:
+                            safe_regex:
+                              regex: "^/api/migration-assessment/api/v1/sources/([0-9a-f-]+)/status$"
+                            headers:
+                            - name: ":method"
+                              exact_match: "PUT"
                           route:
                             cluster: backend-agent-api
-                    http_filters:
-                    - name: envoy.filters.http.router
-                      typed_config:
-                        "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
-  
-          # This listener is used to accept inbound API requests for backend-image.
-          - name: ingress-backend-image
-            address:
-              socket_address:
-                address: 0.0.0.0
-                port_value: 11443
-                protocol: TCP
-            filter_chains:
-              - filters:
-                - name: envoy.filters.network.http_connection_manager
-                  typed_config:
-                    "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
-                    use_remote_address: true # enable x-forward-for
-                    skip_xff_append: false
-                    xff_num_trusted_hops: 1
-                    stat_prefix: backend-image
-                    access_log:
-                    - name: envoy.access_loggers.file
-                      typed_config:
-                        "@type": type.googleapis.com/envoy.extensions.access_loggers.file.v3.FileAccessLog
-                        path: /dev/stdout
-                    route_config:
-                      name: http_route
-                      virtual_hosts:
-                      - name: default
-                        domains: 
-                          - "*"
-                        routes:
-                        - match: 
-                            prefix: /api/v1/image
+                            regex_rewrite:
+                              pattern:
+                                regex: "^/api/migration-assessment/api/v1/sources/([0-9a-f-]+)/status$"
+                              substitution: "/api/v1/sources/\\1/status"
+                        - match:
+                            prefix: "/api/migration-assessment/api/v1/agent"
                           route:
-                            cluster: backend-image
-                            timeout: 0s
+                            cluster: backend-agent-api
+                            regex_rewrite:
+                              pattern:
+                                regex: "^/api/migration-assessment/api/v1/agent$"
```

Comment:
this should be like the source:
`^/api/migration-assessment/api/v1/agents/([0-9a-f-]+)/status$`

Reply:
It's actually not needed here, because all /agent endpoints are related to `agent-api`, but some `/sources` endpoints are related to agent and some to service API

Reply:
in that case, you should remove `$`

Reply:
yup, removing.